File: Frontend/angular-app/src/app/modules/admin/repositories/person-customers/commands/create-person-customer-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";
import {PersonCustomer} from "../../../entities/person-customer";

export class CreatePersonCustomerCommand extends IRequest<CreatePersonCustomerCommand, PersonCustomer> {

  public personId!: number;
  public customerTypeBaseId!: number;
  public customerCode!: string;
  public curentExpertId!: number;
  public economicCode!: string;
  public description!: string;

  constructor() {
    super();
  }

  mapFrom(entity: PersonCustomer): CreatePersonCustomerCommand {
    throw new ApplicationError(CreatePersonCustomerCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  mapTo(): PersonCustomer {
    throw new ApplicationError(CreatePersonCustomerCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/personCustomers/add";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(CreatePersonCustomerCommandHandler.name)
export class CreatePersonCustomerCommandHandler implements IRequestHandler<CreatePersonCustomerCommand, PersonCustomer> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,

  ) {
  }

  async Handle(request: CreatePersonCustomerCommand): Promise<PersonCustomer> {
    let httpRequest: HttpRequest<CreatePersonCustomerCommand> = new HttpRequest<CreatePersonCustomerCommand>(request.url, request);


    return await this._httpService.Post<CreatePersonCustomerCommand, ServiceResult<PersonCustomer>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()
      return response.objResult
    })

  }
}

File: Backend/Microservices/Admin/Eefa.Admin.Application.Refactor/UseCases/PersonCustomer/Commands/Create/CreatePersonCustomerCommand.cs
using System.Threading;
using System.Threading.Tasks;
using AutoMapper;
using MediatR;

public class CreatePersonCustomerCommand : IRequest<ServiceResult<PersonCustomerModel>>, IMapFrom<Customer>
{
    public int PersonId { get; set; } = default!;
    public int CustomerTypeBaseId { get; set; } = default!;
    public string CustomerCode { get; set; } = default!;
    public int CurentExpertId { get; set; } = default!;
    public string? EconomicCode { get; set; }
    public string? Description { get; set; }
    public void Mapping(Profile profile)
    {
        profile.CreateMap<CreatePersonCustomerCommand, Customer>();
    }
}

public class CreatePersonCustomerCommandHandler : IRequestHandler<CreatePersonCustomerCommand, ServiceResult<PersonCustomerModel>>
{
    private readonly IUnitOfWork _unitOfWork;
    private readonly IMapper _mapper;

    public CreatePersonCustomerCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
    {
        _unitOfWork = unitOfWork;
        _mapper = mapper;
    }
    public async Task<ServiceResult<PersonCustomerModel>> Handle(CreatePersonCustomerCommand request, CancellationToken cancellationToken)
    {
        Customer entity = _mapper.Map<Customer>(request);

        _unitOfWork.Customers.Add(entity);
        await _unitOfWork.SaveChangesAsync(cancellationToken);

        return ServiceResult.Success(_mapper.Map<PersonCustomerModel>(entity));
    }
}